move/seal/sources/bf_hmac_encryption.move

```diff
@@ -294,6 +296,19 @@ fun verify_derived_key(
     pairing(derived_key, &g2_generator()) == pairing(gid, public_key)
 }
 
+fun assert_unique_u8(v: &vector<u8>) {
+    let n = v.length();
+    let mut i = 0;
+    while (i < n) {
+        let mut j = i + 1;
+        while (j < n) {
+            assert!(v[i] != v[j]);
+            j = j + 1;
+        };
+        i = i + 1;
+    };
+}
```

Comment:
I think this can be made a bit simpler using macros. Something like:
```suggestion
fun assert_unique<T: drop + copy>(items: &vector<T>) {
    let mut seen = vector::empty();
    items.do_ref!(|item| {
        assert!(seen.find_index!(|i| i == item).is_none());
        seen.push_back(*item);
    })
}
```